src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -226,8 +290,13 @@ private ScheduledInterviewRequest makeScheduledInterviewRequest(
             .get(scheduledInterview.intervieweeId())
             .map(Person::firstName)
             .orElse("Nonexistent User");
-    String shadow =
-        personDao.get(scheduledInterview.shadowId()).map(Person::firstName).orElse("None");
+    // When an interview is first scheduled, the shadowId is set to an empty string. Since this
+    // behaviour is expected, here we prevent a null or empty name exception with creating keys
+    // in datastore.
+    String shadow = "None";
+    if (!scheduledInterview.shadowId().equals("")) {
+      shadow = personDao.get(scheduledInterview.shadowId()).map(Person::firstName).orElse("None");
```

Comment:
Kind of clunky to duplicate the "None" thing.

Maybe just:
```java
    if (!scheduledInterview.shadowId().equals("")) {
      Optional<String> firstName = personDao.get(scheduledInterview.shadowId()).map(Person::firstName);
      if (firstName.hasValue()) {
        shadow = firstName.get();
      }
    }
```

Reply:
Done